### ISSUE TITLE
[BUG] Crash with script (invalid memory address or nil pointer dereference)

### ISSUE THREAD

**Issue by dflvunoooooo** [joshuar/go-hass-agent#133]
**Go Hass Agent Version**
go-hass-agent: v7.3.0

**Describe the bug**
If I try to include my script, go-hass-agent will not even start. It does not matter if it is written in bash or go. I get the error 
```
10:17AM DBG Debug logging enabled.
10:17AM DBG Setting language to [de].
10:17AM DBG Agent already registered.
10:17AM DBG Listening for notifications.
10:17AM DBG Starting worker funcs.
10:17AM WRN Cannot run script error="fork/exec /home/peter/.config/com.github.joshuar.go-hass-agent/scripts/system update info.sh: exec format error" script="/home/peter/.config/com.github.joshuar.go-hass-agent/scripts/system update info.sh"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xd4e178]

goroutine 80 [running]:
github.com/joshuar/go-hass-agent/internal/scripts.(*script).Schedule(...)
        /home/joshua/git/go-hass-agent/internal/scripts/scripts.go:63
github.com/joshuar/go-hass-agent/internal/agent.runScripts({0x1161720, 0xc00036b080}, {0xc0001f5080?, 0x1d49c0001d456?}, {0x1161560, 0xc000396780}, {0x1161598, 0xc000113c20})
        /home/joshua/git/go-hass-agent/internal/agent/runners.go:90 +0x158
github.com/joshuar/go-hass-agent/internal/agent.(*Agent).Run.func2.3()
        /home/joshua/git/go-hass-agent/internal/agent/agent.go:96 +0x145
created by github.com/joshuar/go-hass-agent/internal/agent.(*Agent).Run.func2 in goroutine 69
        /home/joshua/git/go-hass-agent/internal/agent/agent.go:93 +0x248
```

**To Reproduce**
Steps to reproduce the behaviour:
1. copy script in config
2. make it executable
3. start go-hass-agent

**Expected behaviour**
I would expect it to start and run the program.

**Logs**
[go-hass-agent.log](https://github.com/joshuar/go-hass-agent/files/14976102/go-hass-agent.log)

**Desktop (please complete the following information):**
 - OS: Linux
 - Distribution Arch Linux
 - Version Linux 6.8.2-arch1-1

**Additional context**
Here is my script in go and in bash

[system update info.sh.txt](https://github.com/joshuar/go-hass-agent/files/14976115/system.update.info.sh.txt)
[system_update_info.go.txt](https://github.com/joshuar/go-hass-agent/files/14976117/system_update_info.go.txt)

**Comment by joshuar** [comment]
Hey @dflvunoooooo it looks like the script here does not contain a valid shebang line. The scripts need to be able to be run directly on the command-line when invoked (i.e. `./system_update_info.sh` should work). I think if you add a valid shebang line (like `#!/bin/sh`) as the first line, this script will work?

It is a problem that the agent crashed with a panic though. I'll look into fixing that.

**Comment by dflvunoooooo** [comment]
Thank you for answering. I seem to have accidentally deleted the shebang. I now can run the script from the command line the way you mentioned. But I have the same error after adding it again.

**Comment by joshuar** [comment]
Okay I have been able to reproduce the crash on my Arch VM, it is fixed by 07d3e0ea36a4a3836641e4585964a4278bbed62a, which will be included in the next release, to happen imminently :tada: 

I did however also encounter some JSON parsing issues with your script output. Here is a modified version that works for me:

```bash
#!/bin/bash

update_seconds=$((($(date +%s) - $(date -d $(sed -n '/upgrade$/x;${x;s/.\([0-9-]*\).*/\1/p}' /var/log/pacman.log) +%s))))
number_updates=$(($(sudo yay -Sy > /dev/null && yay -Qu | wc -l) - 1))

# JSON-Format ausgeben
cat << EOF
{
  "schedule": "@every 12h",
  "sensors": [
    {
      "sensor_name": "System Last Update",
      "sensor_icon": "mdi:update",
      "sensor_state": $update_seconds,
      "sensor_device_class": "duration",
      "sensor_units": "s"
    },
    {
      "sensor_name": "System Available Updates",
      "sensor_icon": "mdi:numeric",
      "sensor_state": $number_updates
    }
  ]
}
EOF
```

I've cleaned up the JSON and used a [HEREDOC](https://tldp.org/LDP/abs/html/here-docs.html) to make writing the JSON output a little easier (without needing to escape a lot of strings). 

You can use [dasel](https://github.com/TomWright/dasel) to [validate](https://daseldocs.tomwright.me/v/v1/usage/validate) the JSON output of the script. Once you have dasel installed, a command like:

```shell
~/.config/com.github.joshuar.go-hass-agent/scripts/system.update.info.sh > /tmp/script.output.json && dasel validate /tmp/script.output.json && rm /tmp/script.output.json
```

Should print out `pass /tmp/script.output.json` or a `fail (some error)...` if it failed to validate.
